sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/GlobalEndpointManager.java

```diff
@@ -41,6 +41,7 @@
     private final ConnectionPolicy connectionPolicy;
     private final DatabaseAccountManagerInternal owner;
     private final AtomicBoolean isRefreshing;
+    private final AtomicBoolean refreshInBackground;
```

Comment:
why can't we reuse isRefreshing? what's the difference between isRefreshing and refreshInBackground?

Reply:
That's the main part of this PR , isRefreshing -> if there is region outage then all the requests should not start account refresh and only one do the work . refreshInBackground -> this is for the background periodic task. 
Initially both the activities were govern by single flag and causing overlaps and  issues.
Please connect with me offline  I will walk you through the code and logic